Core/include/Acts/EventData/GenericBoundTrackParameters.hpp

```diff
@@ -41,6 +42,76 @@ class GenericBoundTrackParameters {
   using CovarianceMatrix = BoundSquareMatrix;
   using ParticleHypothesis = particle_hypothesis_t;
 
+  /// Factory to construct from four-position, direction, absolute momentum, and
+  /// charge.
+  ///
+  /// @param surface Reference surface the parameters are defined on
+  /// @param geoCtx Geometry context for the local-to-global transformation
+  /// @param pos4 Track position/time four-vector
+  /// @param dir Track direction three-vector; normalization is ignored
+  /// @param qOverP Charge over momentum
+  /// @param cov Bound parameters covariance matrix
+  /// @param particleHypothesis Particle hypothesis
+  /// @param tolerance Tolerance used for globalToLocal
+  ///
+  /// @note The returned result indicates whether the free parameters could
+  /// successfully be converted to on-surface parameters.
+  static Result<GenericBoundTrackParameters> create(
+      std::shared_ptr<const Surface> surface, const GeometryContext& geoCtx,
+      const Vector4& pos4, const Vector3& dir, double qOverP,
+      std::optional<CovarianceMatrix> cov,
+      ParticleHypothesis particleHypothesis,
+      double tolerance = s_onSurfaceTolerance) {
+    Result<BoundVector> bound =
+        transformFreeToBoundParameters(pos4.segment<3>(ePos0), pos4[eTime], dir,
+                                       qOverP, *surface, geoCtx, tolerance);
+
+    if (!bound.ok()) {
+      return bound.error();
+    }
+
+    return GenericBoundTrackParameters{std::move(surface), std::move(*bound),
+                                       std::move(cov),
+                                       std::move(particleHypothesis)};
+  }
+
+  /// Construct from four-position, direction, and qOverP.
+  ///
+  /// @param pos4 Track position/time four-vector
+  /// @param dir Track direction three-vector; normalization is ignored.
+  /// @param qOverP Charge over momentum
+  /// @param cov Curvilinear bound parameters covariance matrix
+  /// @param particleHypothesis Particle hypothesis
+  static GenericBoundTrackParameters makeCurvilinear(
+      const Vector4& pos4, const Vector3& dir, double qOverP,
+      std::optional<CovarianceMatrix> cov,
+      ParticleHypothesis particleHypothesis) {
+    return GenericBoundTrackParameters(
+        CurvilinearSurface(pos4.segment<3>(ePos0), dir).surface(),
+        transformFreeToCurvilinearParameters(pos4[eTime], dir, qOverP),
+        std::move(cov), std::move(particleHypothesis));
+  }
+
+  /// Construct from four-position, angles, and qOverP.
+  ///
+  /// @param pos4 Track position/time four-vector
+  /// @param phi Transverse track direction angle
+  /// @param theta Longitudinal track direction angle
+  /// @param qOverP Charge over momentum
+  /// @param cov Curvilinear bound parameters covariance matrix
+  /// @param particleHypothesis Particle hypothesis
+  static GenericBoundTrackParameters makeCurvilinear(
```

Comment:
```suggestion
  static GenericBoundTrackParameters createCurvilinear(
```

Also should this be a separate factory, or do we just require the user to manually call `makeDirectionFromPhiTheta`?

Reply:
I think we can drop it. I just moved it from the previous implementation

Reply:
actually I went back to this because providing `phi, theta` will not need any conversion for the track params. without that unit tests fail